.github/actions/install-aiida-core/action.yml

```diff
@@ -25,26 +25,18 @@ runs:
     with:
       python-version: ${{ inputs.python-version }}
 
-  - name: Install uv installer
-    run: curl --proto '=https' --tlsv1.2 -LsSf https://${{ env.UV_URL }} | sh
-    env:
-      UV_VERSION: 0.2.9
-      UV_URL: github.com/astral-sh/uv/releases/download/$UV_VERSION/uv-installer.sh
-    shell: bash
+  - name: Set up uv
+    uses: astral-sh/setup-uv@v4
+    with:
+      version: 0.5.5
 
-  - name: Install dependencies from requirements-py-*.txt
-    if: ${{ inputs.from-requirements == 'true' }}
-    run: uv pip install --system -r requirements/requirements-py-${{ inputs.python-version }}.txt
+  - name: Install dependencies from uv lock
+    # NOTE: We're also asserting that the lockfile is up to date
+    if: ${{ inputs.from-lock == 'true' }}
+    run: uv sync --locked --extra pre-commit
```

Comment:
NOTE: By default we're installing with `pre-commit` optional dependencies (extras), which transitively includes other extras, see pyproject.toml I think this is the best solution to make things work for now.

Reply:
I added a comment for now. I agree this is not ideal, I have some ideas but want to keep them for a separate PR since there's a lot going on here already.

Note that if we wanted to always install all the extras, we could just pass the `--all-extras` parameter to uv. But that would (among others) install the whole jupyter stack which seems wasteful.